proxy/envoy/ingress.go

```diff
@@ -111,8 +130,38 @@ func generateIngress(conf *IngressConfig) *Config {
 		rulesByHost[host] = append(rulesByHost[host], rule)
 	}
 
-	// Phase 2: create a VirtualHost for each host
+	// figure out which hosts are configured to terminate TLS
+	// ensure that only one TLS config is used
+	// TODO: extensive tests
+	tlsValid := true
+	tls, err := conf.Secrets.GetSecret("*")
+	if err != nil {
+		glog.Warningf("Error retrieving TLS context for wildcard: %v", err)
+		tlsValid = false
+	}
+	wildcardTLS := tls != nil
+
+	tlsHosts := make(map[string]bool)
+	for host := range rulesByHost {
+		if t, err := conf.Secrets.GetSecret(host); err != nil {
+			glog.Warningf("Error retrieving TLS context for %q: %v", host, err)
+			tlsHosts[host] = true
```

Comment:
I'm missing `tlsValid = false`. If we can't determine whether something is TLS, we omit it from the vhosts that we generate.